packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -454,7 +521,15 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
         child: AutocompleteHighlightedOption(
           highlightIndexNotifier: _highlightedOptionIndex,
           child: Builder(
-            builder: (BuildContext context) => widget.optionsViewBuilder(context, _select, _options),
+            builder: (BuildContext context) {
+              if (_isLoadingOptions && widget.loadingStateBuilder != null) {
+                return widget.loadingStateBuilder!(context, _options);
+              }
+              if (_options.isEmpty && widget.emptyStateBuilder != null) {
+                return widget.emptyStateBuilder!(context, _textEditingController.text);
+              }
+              return widget.optionsViewBuilder(context, _select, _options);
```

Comment:
What's the advantage of separating out loadingStateBuilder and emptyStateBuilder instead of handling everything inside of optionsViewBuilder? Maybe with the addition of an isLoading parameter or something.